teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,398 @@
+    /0.2
+      /README.md
+      /go-release.yaml
+      /samples
+      /tests
+    /OWNERS
+```
+
+For further information, see [TEP-0003: Tekton Catalog Organization][tep-0003].
+
+The current directory-based organization and versioning presents challenges:
+- Bumping versions of resources is cumbersome which encourages making "minor" changes in existing versions. Some changes
+that seem minor cause surprising breaking changes for users, such as described in [tektoncd/catalog#784][784].
+- Resources that are frequently updated with new versions end up with a lot of duplications because any change requires
+copying over everything into a new directory. For example, [git-clone][git-clone] `Task` has seven versions with a lot
+of duplication across versions.
+
+Previously, there was tight coupling between the organization and how users referenced resources which made it hard to 
+change the organization of the Catalog without breaking users. In [TEP-0110][tep-0110], we decoupled the Catalog 
+organization from resource reference. Now, we can create a new organization and versioning contract for Catalogs to
+address the challenges with the directory-based organization and versioning.
+
+The aim of this TEP is to create a new organization and versioning for Tekton Catalogs, building on the prior work and 
+experiences from [TEP-0003][tep-0003]. In [TEP-0079: Tekton Catalog Support Tiers][tep-0079], we plan to add a Tekton 
+Official Catalog which will use this new organization and versioning contract - therefore, this proposal is a blocker 
+for [TEP-0079][tep-0079].
+
+## Proposal
+
+We propose supporting a flat organization and git-based versioning of Tekton Catalogs.
+
+### Flat Organization
+
+In this organization contract, we'll use a flat structure instead of the directory structure. The example shown above
+with the two `Tasks` (argocd and golang-build) and one `Pipeline` (go-release) would be organized as such:
+
+```shell
+./task/
+  /argocd
+    /README.md
+    /argocd.yaml
+    /samples
+    /tests
+    /OWNERS
+  /golang-build
+    /README.md
+    /golang-build.yaml
+    /samples
+    /tests
+    /OWNERS
+./pipeline/
+  /go-release
+    /README.md
+    /go-release.yaml
+    /samples
+    /tests
+    /OWNERS
+```
+
+Note that there are no changes to the contents of the resources themselves, only the organization of Catalogs.
+
+### Git-Based Versioning
+
+#### Versioning via Tags
+
+Versioning of resources will be done through creating tags in the Catalogs. The versions in the Tekton Hub should
+continue to use the simple version of [semver][semver]: `{major}.{minor}`. We propose that the tags use the semantic
+versioning model - `v<major-version>.<minor-version>`. Minor versions are made to signal backwards-compatible changes,
+while major versions are made to signal backwards-incompatible changes. As such, the tags in the catalogs will map to
+versions in the Hub e.g. tag "v0.7" in a Catalog maps to version "0.7" in the Hub.
```

Comment:
Do we want to enforce that Hub will only recognize the above versioning model - `v<major-version>.<minor-version>`? I think we need to document it clearly here if that is the case.

/lgtm otherwise.